pkg/events/events.go

```diff
@@ -12,6 +14,7 @@ import (
 type Event interface {
 	Emit() ([]byte, error)
 	String() string
+	LogWith() log.Logger
```

Comment:
`xxWith()` implies taking a parameter to me... I think a better name for this would be `Logger()`.